tests/RobotFramework/tests/cumulocity/software_management/software.robot

```diff
@@ -125,6 +126,33 @@ Workflow log includes plugin output
     ${log_output}=     Execute Command    cat ${operation_log_file}
     Should Contain    ${log_output}    Executing command: "apt-get" "--quiet" "--yes" "update"
 
+Backward compatibility using 501-503 templates to update status
+    Execute Command    tedge config set c8y.smartrest.with_operation_id false
+    Restart Service    tedge-mapper-c8y
+    Service Health Status Should Be Up    tedge-mapper-c8y
+    ${OPERATION}=    Install Software        c8y-remote-access-plugin
+    Operation Should Be SUCCESSFUL           ${OPERATION}    timeout=60
+    Device Should Have Installed Software    c8y-remote-access-plugin
+    Should Have MQTT Messages    c8y/s/us    message_pattern=^(501|502|503),c8y_SoftwareUpdate.*    minimum=1
+
+Operation gets updated regardless of the order of the creation time
+    Stop Service    tedge-agent
+    ${OPERATION1}=    Install Software        non-existent-package1
+    ${OPERATION2}=    Install Software        non-existent-package2
+    ${op1dict}=    Operation Should Be PENDING    ${OPERATION1}
+    ${op2dict}=    Operation Should Be PENDING    ${OPERATION2}
+    ${op_id1}=    Get From Dictionary    ${op1dict}    id
```

Comment:
The Keyword Get From Dictionary is expecting the optional value. If optional default value is given, it will be returned instead of failing.
so the step should look like:

${op_id1}=    Get From Dictionary    ${op1dict}    id    default

the step is passing even if this is not defined, difference is that the keyword is expecting it and underlined red as notcomplete/faulty

Reply:
see below my comment